DUI3-DX/Connectors/Revit/Speckle.Connectors.RevitShared/Plugin/DllConflictManagment/DllConflictManager.cs

```diff
@@ -0,0 +1,245 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Autodesk.Revit.UI;
+
+namespace Speckle.Connectors.Revit.Plugin.DllConflictManagment;
+
+public class DllConflictManager
+{
+  private readonly Dictionary<string, AssemblyConflictInfo> _assemblyConflicts = new();
+  private readonly DllConflictManagmentOptionsLoader _optionsLoader;
+
+  public DllConflictManager(DllConflictManagmentOptionsLoader optionsLoader)
+  {
+    _optionsLoader = optionsLoader;
+  }
+
+  public void DetectConflicts()
+  {
+    Dictionary<string, Assembly> loadedAssembliesDict = new();
+    foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+    {
+      // TODO : what to do about multiple versions of loaded dlls?
+      loadedAssembliesDict[assembly.GetName().Name] = assembly;
+    }
+    LoadAssemblyAndDependencies(typeof(DllConflictManager).Assembly, loadedAssembliesDict, new HashSet<string>());
+  }
+
+  private void LoadAssemblyAndDependencies(
+    Assembly assembly,
+    Dictionary<string, Assembly> loadedAssemblies,
+    HashSet<string> visitedAssemblies
+  )
+  {
+    if (visitedAssemblies.Contains(assembly.GetName().Name))
```

Comment:
So if we have this assembly by Name and a match, we are happy there's no conflict - Does the Name contain the verson always? Or is this something else.... ?